File: p360/p360_43/node/src/services/dataService.js
// Service layer with data encryption
const crypto = require('crypto');
const config = require('../config/security');

class DataService {
    encryptData(data) {
        const cipher = crypto.createCipher('aes-256-cbc', config.encryptionKey);
        let encrypted = cipher.update(JSON.stringify(data), 'utf8', 'hex');
        encrypted += cipher.final('hex');
        return encrypted;
    }

    decryptData(encryptedData) {
        const decipher = crypto.createDecipher('aes-256-cbc', config.encryptionKey);
        let decrypted = decipher.update(encryptedData, 'hex', 'utf8');
        decrypted += decipher.final('utf8');
        return JSON.parse(decrypted);
    }
}

module.exports = new DataService();

File: p360/p360_43/java/src/main/java/com/company/service/DataService.java
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class DataService {

    @Autowired
    private EncryptionService encryptionService;

    public String encryptData(String data) {
        return encryptionService.encrypt(data);
    }

    public String decryptData(String encryptedData) {
        return encryptionService.decrypt(encryptedData);
    }
}